registry/endpoint.go

```diff
@@ -171,12 +168,12 @@ func (e *Endpoint) pingV1(factory *requestdecorator.RequestFactory) (RegistryInf
 		return RegistryInfo{Standalone: false}, nil
 	}
 
-	req, err := factory.NewRequest("GET", e.Path("_ping"), nil)
+	req, err := http.NewRequest("GET", e.Path("_ping"), nil)
 	if err != nil {
 		return RegistryInfo{Standalone: false}, err
 	}
 
-	resp, _, err := doRequest(req, nil, ConnectTimeout, e.IsSecure)
+	resp, err := HTTPClient(DefaultTransport(ConnectTimeout, e.IsSecure)).Do(req)
```

Comment:
doRequest instantiates a new client everysingle time which is weird indeed. Initially i made it so that there was only one client, but I'm not 100% sure it's a good idea, either way thought it's out of scope unless something is actually broken.

However, everything in session.go reuses the same client embedded in the Session struct. That is so that the same cookiejar is reused, like it was the case on master by passing the jar around every function. On this line `doRequest` has a `nil` jar, which is what `HTTPClient()` does by default.

I'd be happy to revisit this in another PR. (again: unless there is something fundamentally wrong)